tests/unit/test_dns.cpp

```diff
@@ -38,6 +38,34 @@
 
 namespace ot {
 
+void TestDnsNameMatchDomain(void)
+{
+    const char *name;
+    const char *domain;
+
+    // Verify it works for the positive case.
+    {
+        name   = "my-service._ipps._tcp.local.";
+        domain = "local.";
+        VerifyOrQuit(Dns::Name::MatchDomain(name, domain), "Name::MatchDomain() failed");
+
+        name   = "my-service._ipps._tcp.default.service.arpa.";
+        domain = "default.service.arpa.";
+        VerifyOrQuit(Dns::Name::MatchDomain(name, domain), "Name::MatchDomain() failed");
+
+        name   = "my-service._ipps._tcp.default.service.arpa.";
+        domain = "service.arpa.";
+        VerifyOrQuit(Dns::Name::MatchDomain(name, domain), "Name::MatchDomain() failed");
+    }
+
+    // Verify it doesn't match a portion of a label.
+    {
+        name   = "my-service._ipps._tcp.default.service.arpa.";
+        domain = "vice.arpa.";
+        VerifyOrQuit(!Dns::Name::MatchDomain(name, domain), "Name::MatchDomain() succeed");
+    }
+}
+
```

Comment:
Should call the newly added `TestDnsNameMatchDomain` from `main()` below?

Reply:
Actually I think it may be better if we add this code directly in `TestDnsName()` function itself? 
It's the function that tests all other methods in `Dns::Name` so makes sense for it to cover the `MatchDomain` as well.

Thoughts?

Reply:
Thanks, I moved it to `TestDnsName()`.